benchmarks/yamauchi_takei_2016_anelasticity/anelasticity_temperature.cc

```diff
@@ -0,0 +1,469 @@
+				const double activation_volume,
+				const double solidus_gradient,
+				const bool use_original_model) const
+	{
+	  // initialize lab-calibrated parameters
+	  const double critical_homologous_temperature = 0.94;
+	  const double reduction_factor = 5;
+	  const double background_amplitude = 0.664;
+	  const double background_slope = 0.38;
+	  const double peak_period = 6e-5;
+	  const double melt_viscosity_factor = 0;
+	  const double melt_peak_factor = 0;
+	  const double reference_temperature = 1473;
+	  const double reference_pressure = 1.5e9;
+	  const double grain_size = 1e-3;
+	  const double reference_grain_size = 1e-3;
+	  const double grain_size_exponent = 3;
+	  const double pressure_gradient = 3e-5;
+	  const double gas_constant=8.3145;
+	  // initialize density parameterisation
+	  const double a=1;
+	  const double b=3;
+	  const double tol=1.48e-8;
+	  int const max_iter=500;
+	  const double bulk_modulus=130e9;
+	  const double bulk_modulus_pressure_derivative=4.8;
+	  const double gruneisen_parameter=6;
+	  const double reference_density=3330;
+	  // initialize original density parameterisation
+	  const double original_density=3291;
+	  const double original_thermal_expansivity=3.59e-5;
+	  const double original_bulk_modulus=115.2;
+	  // initialize solidus
+	  const double T_solidus = 1326.0 + 273 + (((depth-50000)/1e3)*solidus_gradient);
+	  // initialize homologous_temperature
+	  double homologous_temperature = temperature/T_solidus;
+	  // initialize pressures
+	  double pressure = depth/pressure_gradient;
+	  // initialize other parameters
+	  double viscosity,viscosity_reduction_factor,peak_amplitude,peak_width,isothermal_volume_change;
+	  double compressibility,pressure_dependent_density,integrated_thermal_expansivity,density,anharmonic_Vs;
+	  double unrelaxed_compliance,loss_compliance,storage_compliance,attenuation,period,anelastic_Vs;
+
+	  // begin calculation of Vs
+	  if (homologous_temperature<critical_homologous_temperature)
+	    {
+	      viscosity_reduction_factor=1;
+	    }
+	  else if ((homologous_temperature>=critical_homologous_temperature) && (homologous_temperature<1))
+	    {
+	      viscosity_reduction_factor=exp((-1*((homologous_temperature-critical_homologous_temperature)/(homologous_temperature-
+			  (homologous_temperature*critical_homologous_temperature))))*log(reduction_factor));
+		}
+	  else
+	    {
+		  viscosity_reduction_factor=(1/reduction_factor)*exp(-melt_viscosity_factor);
+	    }
+	  viscosity = pow(grain_size/reference_grain_size,grain_size_exponent)*viscosity_prefactor*exp((activation_energy/gas_constant)
+		  *(1/temperature-1/reference_temperature))*exp((activation_volume/gas_constant)*(pressure/temperature-reference_pressure/
+			  reference_temperature))*viscosity_reduction_factor;
+	  unrelaxed_compliance=1/(1e9*(mu0+(dmudP*pressure*1e-9)+(dmudT*(temperature-273))));
+	  if ((temperature-273<0) || (viscosity > 1e40))
+	    {
+		  temperature=273;
```

Comment:
you want to reset both even if one is invalid?

Reply:
I have changed this now, I only really want to change these things if temperature is less than 273